src/main/java/org/openrefine/extensions/commons/utils/CommonsImportingController.java

```diff
@@ -206,60 +207,67 @@ static public void parse(
 
         JSONUtilities.safePut(options, "headerLines", 0);
         /* get user-input from the Post request parameters */
-        String cmtitle = JSONUtilities.getString(options, "categoryInput", null);
-        if (!cmtitle.startsWith("Category:")) {
-            cmtitle = "Category:" + cmtitle;
+        JsonNode categoryInput = options.get("categoryJsonValue");
+        List<String> categories = new ArrayList<>();
+        for (JsonNode category: categoryInput) {
+            categories.add(category.get("category").asText());
         }
-        String pageName = cmtitle;
-        String fileSource = pageName;//FIXME: add filename
         String apiUrl = "https://commons.wikimedia.org/w/api.php";
 
-        setProgress(job, fileSource, 0);
+        setProgress(job, categories.get(0), 0);// handles the first category
```

Comment:
Maybe the comment is a bit misleading, no? That line in itself does not "handle" the first category: it just sets up the progress reporting by initializing it to contain the name of the first category.